### PR TITLE
Toxic Exposure - Data Transformation #3 - Add Conditions

### PR DESCRIPTION
**Pull request by sethdarragile6**
This PR maps transforms Lighthouse request disability attributes based on new toxic exposure information, namely:
- `isRelatedToToxicExposure` (from matching to the `toxicExposure/conditions` collection)
- `exposureOrEventOrInjury` (from the FE caption label for `cause` ("NEW", "WORSENED", "SECONDARY", "VA")

## Summary

- *This work is behind a feature toggle (flipper): YES* `disability_526_toxic_exposure`
- *(Which team do you work for, does your team own the maintenance of this component?)*
   Disability Benefits Experience Team 1 (DBEX-TREX 🦖)

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
   Toxic Exposure - Data Transformation #3 - Add Conditions#[81426](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/81426)

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by aurora-a-k-a-lightning — DISMISSED** [review]

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 1]
```diff
@@ -307,22 +317,37 @@ def convert_approximate_date(approximate_date_source, short: false)
...
+          if disability_source['cause'].present?
+            dis.exposure_or_event_or_injury = TOXIC_EXPOSURE_CAUSE_MAP[disability_source['cause'].upcase.to_sym]
+          end
+
           dis
         end
       end
 
+      # rubocop:disable Naming/PredicateName
+      def is_related_to_toxic_exposure(condition_name, toxic_exposure_conditions)
```
[the remark this answers is not in the document]
Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:344]: agreed. just trying to keep some semblance of parity between LH, BE and FE naming. good luck with that, right? 😂

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 2]
```diff
@@ -5,6 +5,15 @@
...
   module DisabilityCompensationForm
     class Form526ToLighthouseTransform
+      TOXIC_EXPOSURE_CAUSE_MAP = {
+        NEW: 'My condition was caused by an injury or exposure during my military service.',
+        WORSENED: 'My condition existed before I served in the military, but it got worse because of my military ' \
+                  'service.',
+        VA: 'My condition was caused by an injury or event that happened when I was receiving VA care.',
+        SECONDARY: 'My condition was caused by another service-connected disability I already have. (For example, I ' \
+                   'have a limp that caused lower-back problems.)'
+      }.freeze
```
Comment (christinec-fftc) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:15]: Not a big deal, but i wonder if the VSR's need the example text.

Comment (sethdarragile6) [reply lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:14]: Good point. Got the go-ahead from Shannon to shorten it. 👍

**Review by aurora-a-k-a-lightning — APPROVED** [review]

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 3]
```diff
@@ -307,22 +317,37 @@ def convert_approximate_date(approximate_date_source, short: false)
         approximate_date
       end
 
-      def transform_disabilities(disabilities_source)
+      def transform_disabilities(disabilities_source, toxic_exposure_conditions)
         disabilities_source.map do |disability_source|
           dis = Requests::Disability.new
           dis.disability_action_type = disability_source['disabilityActionType']
           dis.name = disability_source['name']
+          if toxic_exposure_conditions.present? && toxic_exposure_conditions.any?
```
Comment (freeheeling) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:324]: Could this be simplified as `if toxic_exposure_conditions&.any?`

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 4]
```diff
@@ -22,7 +31,8 @@ def transform(evss_data)
         lh_request_body.service_information = transform_service_information(service_information)
 
         disabilities = form526['disabilities']
-        lh_request_body.disabilities = transform_disabilities(disabilities)
+        toxic_exposure_conditions = form526['toxicExposure']['conditions'] if form526['toxicExposure'].present?
```
Comment (freeheeling) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:33]: Can be simplified as `form526['toxicExposure']&.[]('conditions')`

File: spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb [hunk 5]
```diff
@@ -197,6 +201,23 @@
       result = transformer.send(:convert_approximate_date, JSON.parse({ month: '03', year: '1973' }.to_json))
       expect(result).to eq('1973-03')
     end
+
+    it 'sets the is related to toxic exposure flag when matching to TE conditions' do
```
Comment (freeheeling) [inline spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb:205]: For readability, perhaps add underscores to `is related to toxic exposure`.

File: spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb [hunk 6]
```diff
@@ -197,6 +201,23 @@
...
+      toxic_exposure_conditions = submission.form['form526']['form526']['toxicExposure']['conditions']
+      results = transformer.send(:transform_disabilities, data, toxic_exposure_conditions)
```
Comment (freeheeling) [inline spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb:207]: Could these repeated variables, `toxic_exposure_conditions` and `results`, instead be defined as memoized helper methods?

**Review by freeheeling — COMMENTED** [review]
Some refactor suggestions for your consideration.

**Pull request merged**